pkg/controller/cluster/cluster_reconcile.go

```diff
@@ -63,6 +68,21 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 	repoName := request.NamespacedName
 	repoName.Name = instance.Spec.TenantRef.Name
 
+	client, err := vault.NewClient()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	token, err := r.getServiceAccountToken(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = client.SetToken(path.Join(instance.Spec.TenantRef.Name, instance.Name), token, reqLogger)
```

Comment:
The token should be saved in `<tenantId>/<clusterId>/steward` with `token=<token>`
```suggestion
	err = client.SetToken(path.Join(instance.Spec.TenantRef.Name, instance.Name, "steward"), token, reqLogger)
```